bin/docker-build-proxy

```diff
@@ -13,5 +13,14 @@ rootdir="$( cd $bindir/.. && pwd )"
 . $bindir/_docker.sh
 . $bindir/_tag.sh
 
+# The proxy's build stage may be useful to preserve, especially in CI. We tag this stage
+# separately. The subsequent runtime build is trivial.
+docker_build proxy-build cache $rootdir/proxy/Dockerfile \
+    --build-arg="PROXY_UNOPTIMIZED=${PROXY_UNOPTIMIZED:-}" \
+    --cache-from="$(docker_repo proxy-build):cache" \
```

Comment:
I'm not convinced this is the right behavior outside of Travis CI runs due to https://github.com/moby/moby/issues/34715. My understanding is that if we specify `--cache-from` here then that will disable any/all implicit local caching. IIUC, without `--cache-from`, I will use many different cached versions of the build container locally, as needed for the different branches I have, but with `--cache-from` I will only be able to used one cached version. In particular, if I am switching between different branches with `git checkout` on my local machine that require a different build image than the one specified in `--cache-from` above, then no caching will be done. So, unless I'm misunderstanding, I think we should only do this `--cache-from` in CI and not by default.

Reply:
yeah, this makes sense.